bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/StartupPreferencePage.java

```diff
@@ -73,8 +92,66 @@ protected Control createContents(Composite parent) {
 		return composite;
 	}
 
-	protected void createExtraContent(@SuppressWarnings("unused") Composite composite) {
-		// subclasses may override
+	protected void createExtraContent(Composite composite) {
+		if (WindowsDefenderConfigurator.isRelevant()) {
+			new Label(composite, SWT.NONE); // add spacer
+
+			GridDataFactory grapHorizontalSpace = GridDataFactory.swtDefaults().align(SWT.FILL, SWT.BEGINNING)
+					.grab(true, false);
+
+			Group group = GroupFactory.newGroup(SWT.SHADOW_NONE | SWT.H_SCROLL)
+					.text(WorkbenchMessages.WindowsDefenderConfigurator_statusCheck)
+					.layoutData(grapHorizontalSpace.create()).create(composite);
+			GridLayoutFactory.swtDefaults().spacing(5, 10).applyTo(group);
+
+			String infoText = WindowsDefenderConfigurator
+					.bindProductName(WorkbenchMessages.WindowsDefenderConfigurator_exclusionInformation);
+			LabelFactory.newLabel(SWT.WRAP).text(infoText).layoutData(grapHorizontalSpace.create()).create(group);
+
+			Button ignoreAllInstall = createCheckBox(WorkbenchMessages.WindowsDefenderConfigurator_ignoreAllChoice,
+					false, group);
+
+			windowsDefenderIgnore = Map.of(UserScope.INSTANCE, ignoreAllInstall);
+
+			ButtonFactory.newButton(SWT.PUSH)
+					.text(WorkbenchMessages.WindowsDefenderConfigurator_runExclusionFromPreferenceButtonLabel)
+					.font(composite.getFont()).onSelect(e -> {
+						Shell shell = getShell();
+						try {
+							Boolean excluded = WindowsDefenderConfigurator.runCheckEnforced(null);
+							if (excluded == Boolean.FALSE) {
+								// Show a dialog that Defender is inactive to give some kind of feedback
+								MessageDialog.open(MessageDialog.INFORMATION, shell,
+										WorkbenchMessages.WindowsDefenderConfigurator_statusCheck,
+										WorkbenchMessages.WindowsDefenderConfigurator_statusInactive, SWT.NONE);
+							}
+						} catch (CoreException ex) {
+							ErrorDialog.openError(shell, "Windows Defender exclusion check failed", //$NON-NLS-1$
+									"An unexpected error occured while running the Windows Defender exclusion check.", //$NON-NLS-1$
+									ex.getStatus());
+							WorkbenchPlugin.log("Error while running the Windows Defender exclusion check", ex); //$NON-NLS-1$
+						}
+					}).create(group);
+
+			ExpandBar bar = new ExpandBar(group, SWT.NONE);
+			bar.setForeground(bar.getDisplay().getSystemColor(SWT.COLOR_BLACK));
```

Comment:
I am unsure if setting fixed colors causes issues in the dark theme. Did you test also in the dark and the classic theme?

Reply:
Indeed I only tested it with the light/classic theme, but you are right this isn't a general solution.
If the ExpandBar would be used eventually setting the foreground-color should be reverted.
But with https://github.com/eclipse-platform/eclipse.platform.ui/pull/1453#issuecomment-1941707500 this is obsolete.